src/main/kotlin/org/rust/lang/core/types/infer/TypeInferenceWalker.kt

```diff
@@ -1039,8 +1039,12 @@ class RsTypeInferenceWalker(
 
     private fun inferTryExprType(expr: RsTryExpr): Ty {
         val base = expr.expr.inferType() as? TyAdt ?: return TyUnknown
-        // TODO: make it work with generic `std::ops::Try` trait
-        if (base.item != items.Result && base.item != items.Option) return TyUnknown
+        if (base.item != items.Result && base.item != items.Option) {
+            val (implLookup, knownItems) = expr.implLookupAndKnownItems
```

Comment:
`implLookup` and `knownItems` should already be in the scope (TypeInferenceWalker propertives)